js/woo.js

```diff
@@ -0,0 +1,1532 @@
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchTrades', market, params);
+        const method = this.getSupportedMapping (marketType, {
+            'spot': 'v1PublicGetMarketTrades',
+        });
+        const response = await this[method] (this.extend (request, query));
+        //
+        // {
+        //     success: true,
+        //     rows: [
+        //         {
+        //             symbol: "SPOT_BTC_USDT",
+        //             side: "SELL",
+        //             executed_price: 46222.35,
+        //             executed_quantity: 0.0012,
+        //             executed_timestamp: "1641241162.329"
+        //         },
+        //         {
+        //             symbol: "SPOT_BTC_USDT",
+        //             side: "SELL",
+        //             executed_price: 46222.35,
+        //             executed_quantity: 0.0012,
+        //             executed_timestamp: "1641241162.329"
+        //         },
+        //         {
+        //             symbol: "SPOT_BTC_USDT",
+        //             side: "BUY",
+        //             executed_price: 46224.32,
+        //             executed_quantity: 0.00039,
+        //             executed_timestamp: "1641241162.287"
+        //         },
+        //         ...
+        //      ]
+        // }
+        //
+        const resultResponse = this.safeValue (response, 'rows', {});
+        return this.parseTrades (resultResponse, market, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        //
+        // ### public/market_trades
+        // {
+        //     symbol: "SPOT_BTC_USDT",
+        //     side: "SELL",
+        //     executed_price: 46222.35,
+        //     executed_quantity: 0.0012,
+        //     executed_timestamp: "1641241162.329"
+        // },
+        //
+        // ### fetchOrderTrades, fetchOrder->Transactions[]
+        //
+        // {
+        //     id: '99119876',
+        //     symbol: 'SPOT_WOO_USDT',
+        //     fee: '0.0024',
+        //     side: 'BUY',
+        //     executed_timestamp: '1641481113.084',
+        //     order_id: '87001234',
+        //     order_tag: 'default', <-- this param only in "fetchOrderTrades"
+        //     executed_price: '1',
+        //     executed_quantity: '12',
+        //     fee_asset: 'WOO',
+        //     is_maker: '1'
+        // }
+        //
+        const isFromFetchOrder = ('id' in trade);
+        const timestamp = this.safeTimestamp (trade, 'executed_timestamp');
+        const marketId = this.safeString (trade, 'symbol');
+        if (market === undefined) {
+            market = this.safeMarket (marketId, market, '_');
+        }
+        const symbol = market['symbol'];
```

Comment:
Let's replace the above three lines 389-391 with a one-liner:
```JavaScript
        if (market === undefined) {
            market = this.safeMarket (marketId, market, '_');
        }
```
↓
```JavaScript
        market = this.safeMarket (marketId, market, '_');
```

Reply:
Also, a delimiter will not take the effect here, cause it will only be recognized for 2-word symbols with one delimiter in them. In this case it's 3 words with 2 delimiters.